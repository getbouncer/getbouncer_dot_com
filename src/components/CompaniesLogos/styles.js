import styled from '@emotion/styled'
import { breakpoints, theme } from '~styles'

export const LogosList = styled.ul`
  align-items: center;
  display: flex;
  justify-content: space-between;
  max-width: ${theme.maxWidth};

  @media ${breakpoints.smartphone} {
    flex-wrap: wrap;
    justify-content: center;
    align-items: center;
  }
`

export const Logo = styled.li`
  margin: 20px;

  img {
    display: block;
    height: 30px;
    filter: grayscale(100%);
  }

  @media ${breakpoints.smartphone} {
    display: inline-block;
    width: 33.33%;
  }
`
