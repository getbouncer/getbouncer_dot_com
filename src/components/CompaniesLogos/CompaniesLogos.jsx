import React from 'react'
import { useStaticQuery, graphql } from 'gatsby'
import { LogosList, Logo } from './styles'

const CompaniesLogos = () => {
  const data = useStaticQuery(graphql`
    query LogosQuery {
      prismicGlobals {
        data {
          companies_logo {
            company_logo {
              alt
              url
            }
          }
        }
      }
    }
  `)
  const { companies_logo: companiesLogos } = data.prismicGlobals.data

  return (
    <LogosList>
      {companiesLogos.map(item => (
        <Logo key={item.company_logo.url}>
          <img src={item.company_logo.url} alt={item.company_logo.alt} />
        </Logo>
      ))}
    </LogosList>
  )
}

export default CompaniesLogos
