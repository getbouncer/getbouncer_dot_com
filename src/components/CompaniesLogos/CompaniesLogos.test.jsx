import React from 'react'
import renderer from 'react-test-renderer'
import { describe, expect, it } from 'jest'
import CompaniesLogos from './CompaniesLogos'

describe('CompaniesLogos', () => {
  it('renders correctly', () => {
    const logos = {
      companies_logo: [
        {
          company_logo: {
            alt: 'logo 1',
            url: 'https://www.example.com/logo_1.svg',
          },
        },
        {
          company_logo: {
            alt: 'logo 2',
            url: 'https://www.example.com/logo_2.svg',
          },
        },
      ],
    }
    const tree = renderer
      .create(<CompaniesLogos logos={logos} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
