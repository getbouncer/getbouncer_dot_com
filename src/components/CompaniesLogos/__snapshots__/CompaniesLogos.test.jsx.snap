// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompaniesLogos renders correctly 1`] = `
<ul
  className="css-i0unku"
>
  <li
    className="css-ecly76"
  >
    <img
      alt="logo 1"
      src="https://www.example.com/logo_1.svg"
    />
  </li>
  <li
    className="css-ecly76"
  >
    <img
      alt="logo 2"
      src="https://www.example.com/logo_2.svg"
    />
  </li>
</ul>
`;
